### PR TITLE
Docs: Clean up and latest documentation

### PR DESCRIPTION
**Pull request by gamab**
This PR focuses on improving the clarity and usability of the documentation:

* **Better Organization**: Reorganized content to improve flow and readability. Adding more info about the authn and authz specifics in the dedicated packages. Linking to the RBAC plugin example for more details on the single tenant usage of the library.
* **Code Examples**: Added code snippets with context, explaining the purpose of the various library elements, focusing a bit on gRPC interceptors.
* **Authentication / Authorization flows**: Explained the various authentication and authorization flows we have in our Grafana ecosystem.

### CODE REVIEW

File: READMEv2.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+- **Flexible Deployment:** Adapt to our various deployments with ease.
+
+## Deployment Scenarios & Examples
+
+The library leverages JWT (JSON Web Token) for secure communication and authorization, ensuring only authorized entities access resources:
+
+### How it works
+
+1. **Component Identification:** Grafana, applications, and services identify themselves using JWT access tokens.
+2. **Authentication:** Upon receiving requests, services verify the authenticity of the access token and ensure the caller (identified by the "audience" field) is authorized.
```
Comment (drclau) [inline READMEv2.md:35]: Let's reword this to make it clear that `aud` is referring to the target (`callee`?).

**Review by drclau — APPROVED** [review]
LGTM, w/ some minor corrections.

File: READMEv2.md (hunk 1 above)
Comment (gamab) [reply READMEv2.md:35]: Yes absolutely.

File: authz/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+```
+
+## [unstable / under development ] Multi-tenant authz client
+
+<!-- TODO -->
+
+## Namespace access
+
+<!-- TODO -->
```
Comment (gamab) [inline authz/README.md:69]: I'll tackle that separately
